src/lib/middleware.ts

```diff
@@ -9,12 +10,21 @@ import { NextApiRequestCookies } from "next/dist/next-server/server/api-utils";
 import parse from "urlencoded-body-parser";
 import { v4 as uuidv4 } from "uuid";
 import { CacheEntry, FormCacheFactory, IFormCache } from "./formCache";
-import { formSubmissionCookieId } from "./types";
+import { Registration } from "./registration/registration";
+import { acceptRejectCookieId, formSubmissionCookieId } from "./types";
 import { toArray } from "./utils";
 
+export type BeaconsContext = GetServerSidePropsContext & {
```

Comment:
This type made handling updates to the `handlePageRequest` far simpler and I am hoping that we can extend this to eventually simplify `handlePageRequest` in the future.

Mostly inspired by the decorator pattern (I think), and how this has been done in [virtual visits](https://github.com/madetech/nhs-virtual-visit/blob/master/src/middleware/withContainer.js).